### ISSUE TITLE
Improvements to game library

### ISSUE THREAD

**Issue by tannerkrewson** [tannerkrewson/rocketcrab#47]
@LukeGT's brainstorming:

- Should the game selection UI be more game-centric rather than site-centric? Like selecting Spyfall -> netgames.io instead of netgames.io -> Spyfall?
- Sort by popularity
- maybe some "by game" and "by site" tabs at the top of the page is enough

**Comment by tannerkrewson** [comment]
Sort by popularity, along with sort by newest, and stuff like reviews, etc. I think are essential if the library of rocketcrab games grows  even larger. But, those features require a database and possibly even accounts/login etc, which will be a big, risky task. For the immediate future, focus will be on the easy to implement enhancements and fixes with undeniable payoffs, which i am putting into this [milestone](https://github.com/tannerkrewson/rocketcrab/milestone/2). As a compromise, I think #27 and #31 will serve a similar purpose 😄

**Comment by LukeGT** [comment]
sorting by popularity can be done manually every now and again by just looking at analytics. That's what I've done on netgames :)

**Comment by LukeGT** [comment]
I don't think it's necessary with the amount of games on there now, but once the library of games gets big enough to need some kind of searching/filtering, I kinda like how it's done on google maps. It's just a bunch of pills at the top of the screen, tapping the pill applies the filter. They're sorted by what gets used the most so you don't have to scroll sideways too much.

![Screenshot_20200914-130612](https://user-images.githubusercontent.com/1981836/93039613-4fec6180-f68b-11ea-841d-f232b30a5d09.png)

**Comment by tannerkrewson** [comment]
haha true! i think for a platform like rocketcrab, which needs to be fair for all developers, manually curating a popularity list would be risky... developers might perceive preferential treatment, etc. and even with an automated popularity list, we would need to account for naughty developers opening up lots of 🚀 🦀 tabs to try to make their game rise the list etc. etc. 🤒   ... i'll save those issues for the future 😆 

ooh the material design 2 💊 💊 💊 's! yes!!! 😆

**Comment by LukeGT** [comment]
I don't think you can avoid presenting the games in _some_ order though. How are you going to decide what that order should be? I would've thought a well-defined criteria like popularity is a fair way to do it.

I'm hoping that rocket crab's popularity surpasses what someone can fake by opening tabs haha.

**Comment by alenros** [comment]
@LukeGT You underestimate the power of Selenium web automation!

**Comment by LukeGT** [comment]
That's a whole other kettle of fish haha, only trust will remedy that one. I guess you could count IPs, but they could spin up lots of IPs or keep getting new addresses. I feel like any order you choose is either unfair or can be hacked, beyond randomising the order of all the games each refresh.

**Comment by alenros** [comment]
And randomizing is terrible for UX.
I hope RocketCrab sees enough use for us to revisit this.

**Comment by LukeGT** [comment]
To be clear, randomising was not a serious suggestion haha
